examples/CustomCollectionView-Swift/Sample/MosaicCollectionViewLayout.swift

```diff
@@ -0,0 +1,254 @@
+//
+//  MosaicCollectionViewLayout
+//  Sample
+//
+//  Created by Rajeev Gupta on 11/9/16.
+//
+//  Copyright (c) 2014-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+//  FACEBOOK BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+//  ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+//  CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import Foundation
+import UIKit
+import AsyncDisplayKit
+
+protocol MosaicCollectionViewLayoutDelegate: ASCollectionDelegate {
+  func collectionView(_ collectionView: UICollectionView, layout: MosaicCollectionViewLayout, originalItemSizeAtIndexPath: IndexPath) -> CGSize
+}
+
+class MosaicCollectionViewLayout: UICollectionViewFlowLayout {
+  var numberOfColumns: Int
+  var columnSpacing: CGFloat
+  var _sectionInset: UIEdgeInsets
+  var interItemSpacing: UIEdgeInsets
+  var headerHeight: CGFloat
+  var _columnHeights: [[CGFloat]]?
+  var _itemAttributes: [[UICollectionViewLayoutAttributes]]?
+  var _headerAttributes: [UICollectionViewLayoutAttributes]?
+  var _allAttributes: [UICollectionViewLayoutAttributes]?
+  
+  required override init() {
+    self.numberOfColumns = 2
+    self.columnSpacing = 10.0
+    self.headerHeight = 44.0 //viewcontroller
+    self._sectionInset = UIEdgeInsetsMake(10.0, 10.0, 10.0, 10.0)
+    self.interItemSpacing = UIEdgeInsetsMake(10.0, 0, 10.0, 0)
+    super.init()
+    self.scrollDirection = .vertical
+  }
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  public var delegate : MosaicCollectionViewLayoutDelegate?
+  
+  override func prepare() {
+    super.prepare()
+    
+    _itemAttributes = []
+    _allAttributes = []
+    _headerAttributes = []
+    _columnHeights = []
+    
+    var top: CGFloat = 0
+    
+    let numberOfSections: NSInteger = self.collectionView!.numberOfSections
+    
+    for section in 0 ..< numberOfSections {
+      let numberOfItems = self.collectionView?.numberOfItems(inSection: section)
+      
+      top += _sectionInset.top
+      
+      if (headerHeight > 0) {
+        let headerSize: CGSize = self._headerSizeForSection(section: section)
+        
+        let attributes: UICollectionViewLayoutAttributes = UICollectionViewLayoutAttributes.init(forSupplementaryViewOfKind: UICollectionElementKindSectionHeader, with: NSIndexPath.init(row: 0, section: section) as IndexPath)
```

Comment:
You don't need `let attributes: UICollectionViewLayoutAttributes =`, just use `let attributes =`, the type will be automatically inferred by the equality.
You don't need the `.init(...)` you can just use `(...)` for both the `UICollectionViewLayoutAttributes`and the `IndexPath`